src/passes/OptimizeInstructions.cpp

```diff
@@ -3761,7 +3761,11 @@ struct OptimizeInstructions
 
   // Returns true if the given binary operation can overflow. If we can't be
   // sure either way, we return true, assuming the worst.
-  bool canOverflow(Binary* binary) {
+  //
+  // We can check for an unsigned overflow (more than the max number of bits) or
+  // a signed one (where even reaching the sign bit is an overflow, as that
+  // would turn us from positive to negative).
+  bool canOverflow(Binary* binary, bool signed_) {
```

Comment:
Can we do `isSignedOp` inside this function instead of having it take `signed_` as a parameter?